docs/api-documentation.md

```diff
@@ -0,0 +1,370 @@
+# API Documentation
+
+This document provides a brief overview of the different API endpoints, their usage and purpose.
+
+## Authentication
+
+This section lists the authentication related requests such as login.
+
+### Login
+
+Allows the user to login into the portal and receive the auth token from the API. Requires the user email and password values.
+
+**User Login Flow**
+
+Here is a step-by-step breakdown of the user login flow:
+
+1. **Client-Side Request:**
+
+   - The client sends a login request to the server.
+
+2. **AuthenticationController Handling:**
+
+   - The `AuthenticationController` captures the login request through the `POST /login` endpoint.
```

Comment:
Instead of unordered list, please used numbered list. It is helpful in pinpointing numbers in a document.

Reply:
These are the general point mentioned under a particular step the number might not look good in term of formatting as they don't follow any order